### PR TITLE
Task 12

### PR DESCRIPTION
**Pull request by Xodoc**

### CODE REVIEW

**Review by denja2 — CHANGES_REQUESTED** [review]
Move Identity to a separate service

File: ConsoleApp/WebAPI/Program.cs [hunk 1]
```diff
@@ -38,15 +44,69 @@ public void ConfigureServices(IServiceCollection services)
             services.AddRepositories().AddServices().AddAutoMapper();
             services.AddLogging(x => x.AddSerilog());
 
-            services.AddControllers();
+            services.AddControllers()
+                .AddNewtonsoftJson(options => options.SerializerSettings.ReferenceLoopHandling = ReferenceLoopHandling.Ignore);
```
Comment (denja2) [inline ConsoleApp/WebAPI/Program.cs:48]: Is it necessary to add authorization in task 12?

**Review by denja2 — CHANGES_REQUESTED** [review]

File: ConsoleApp/WebAPI/Program.cs (hunk 1 above)
Comment (Xodoc) [reply ConsoleApp/WebAPI/Program.cs]: done

File: ConsoleApp/AuthenticationServer/Certificates/SigningAudienceCertificate.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        private readonly RSA rsa;
+
+        public SigningAudienceCertificate()
+        {
+            rsa = RSA.Create();
+        }
+
+        public SigningCredentials GetAudienceSigningKey()
+        {
+            string privateXmlKey = File.ReadAllText("./Keys/private_key.xml");
```
Comment (denja2) [inline ConsoleApp/AuthenticationServer/Certificates/SigningAudienceCertificate.cs:17]: P3 var

File: ConsoleApp/AuthenticationServer/Controllers/AuthenticationController.cs [hunk 3]
```diff
@@ -8,17 +8,17 @@ namespace AuthenticationServer.Controllers
     [ApiController]
     public class AuthenticationController : ControllerBase
     {
-        private readonly IAuthorizationService _authorizationService;
+        private readonly IAuthenticationService _authenticationService;
```
Comment (denja2) [inline ConsoleApp/AuthenticationServer/Controllers/AuthenticationController.cs:11]: P2 rename author

File: ConsoleApp/AuthenticationServer/appsettings.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "ConnectionStrings": {
+    "DefaultConnection": "Server=(localdb)\\mssqllocaldb;Database=mentorshipdb;Trusted_Connection=True;"
+  },
+  "Jwt": {
+    "Key": "0caebfbd5983024fde00e841734c0762",
```
Comment (denja2) [inline ConsoleApp/AuthenticationServer/appsettings.json:6]: Remove

**Review by denja2 — CHANGES_REQUESTED** [review]

File: ConsoleApp/AuthenticationServer/Controllers/AuthenticationController.cs (hunk 3 above)
Comment (Xodoc) [reply ConsoleApp/AuthenticationServer/Controllers/AuthenticationController.cs:11]: done

**Pull request merged**